src/Symfony/Component/Console/Application.php

```diff
@@ -1017,6 +1024,19 @@ private function findAlternatives($name, $collection)
     public function setDefaultCommand($commandName)
     {
         $this->defaultCommand = $commandName;
+
+        return $this;
+    }
+
+    public function setUniqueCommand($commandName)
+    {
+        if (null !== $this->uniqueCommand) {
```

Comment:
Why not, but if we do that, we are forcing the order. I mean will not able anymore to first set-up the single command name, then register the command.